File: app/src/main/java/com/example/baitap21/MainActivity.java
package com.example.baitap21;

import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.SwitchCompat;
import androidx.constraintlayout.widget.ConstraintLayout;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;
import androidx.core.view.WindowInsetsControllerCompat;

import java.util.Random;

public class MainActivity extends AppCompatActivity {

    private ConstraintLayout constraintLayout;
    private final Random random = new Random();
    private int[] backgrounds;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        hideSystemBars();
        setContentView(R.layout.activity_main);

        constraintLayout = findViewById(R.id.main);
        SwitchCompat changeBackgroundSwitch = findViewById(R.id.change_background_switch);

        backgrounds = new int[]{
                R.drawable.troi, R.drawable.anime, R.drawable.doraemon
        };

        // Đặt hình nền ngẫu nhiên khi khởi động
        changeRandomBackground();

        // Xử lý sự kiện gạt công tắc
        changeBackgroundSwitch.setOnCheckedChangeListener((buttonView, isChecked) -> {
            if (isChecked) {
                changeRandomBackground();
            }
        });
    }

    private void changeRandomBackground() {
        if (backgrounds.length > 0) {
            int randomBackground = backgrounds[random.nextInt(backgrounds.length)];
            constraintLayout.setBackgroundResource(randomBackground);
        }
    }

    private void hideSystemBars() {
        WindowInsetsControllerCompat windowInsetsController =
                ViewCompat.getWindowInsetsController(getWindow().getDecorView());
        if (windowInsetsController != null) {
            windowInsetsController.hide(WindowInsetsCompat.Type.systemBars());
            windowInsetsController.setSystemBarsBehavior(
                    WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE);
        }
        if (getSupportActionBar() != null) {
            getSupportActionBar().hide();
        }
    }
}

File: app/src/main/java/com/example/btapt22/MainActivity.kt
package com.example.btapt22

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.viewpager2.widget.ViewPager2
import androidx.viewpager2.adapter.FragmentStateAdapter

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        // sử dụng layout activity_swipe.xml (chứa ViewPager2 id = viewPager)
        setContentView(R.layout.activity_swipe)

        // toolbar nếu có trong layout
        val toolbar = findViewById<androidx.appcompat.widget.Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)

        val viewPager = findViewById<ViewPager2>(R.id.viewPager)
        viewPager.adapter = object : FragmentStateAdapter(this) {
            override fun getItemCount(): Int = 2
            override fun createFragment(position: Int) =
                if (position == 0) LeftFragment() else RightFragment()
        }
    }
}
